iceoryx_meta/build_options.cmake

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.5)
+
+option(BUILD_ALL                        "Build with all extensions and all tests" OFF)
+option(TOML_CONFIG                      "Activates or deactivates TOML support - without TOML RouDi will not be build" ON)
+option(ONE_TO_MANY_ONLY                 "Restricts communication to 1:n pattern" OFF)
+option(BUILD_STRICT                     "Build is performed with '-Werror'" OFF)
+option(BUILD_TEST                       "Build all tests" OFF)
+option(COVERAGE                         "Build iceoryx with gcov flags" OFF)
+option(EXAMPLES                         "Build all iceoryx examples" OFF)
+option(INTROSPECTION                    "Builds the introspection client which requires the ncurses library with an activated terminfo feature" ON)
+option(DDS_GATEWAY                      "Builds the iceoryx dds gateway - enables internode communication via dds" OFF)
+option(BINDING_C                        "Builds the C language bindings" ON)
+option(SANITIZE                         "Build with sanitizers" OFF)
+option(CLANG_TIDY                       "Execute Clang-Tidy" OFF)
+option(ROUDI_ENVIRONMENT                "Build RouDi Environment for testing, is enabled when building tests" OFF)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)   #"Create compile_commands.json file"
+
+
+if(BUILD_ALL)
+    set(EXAMPLES ON)
+    set(INTROSPECTION ON)
+    set(BUILD_TEST ON)
+    set(INTROSPECTION ON)
+    set(BINDING_C ON)
+    set(DDS_GATEWAY ON)
+endif(BUILD_ALL)
+
+if(BUILD_TEST)
+    set(ROUDI_ENVIRONMENT ON)
+    set(BINDING_C ON)
+endif(BUILD_TEST)
+
+if(EXAMPLES)
+    set(BINDING_C ON)
+endif(EXAMPLES)
+
+message("")
+message("       Configured Options")
+message("          Build Mode........................: " ${CMAKE_BUILD_TYPE})
```

Comment:
Here I would name the options directly. The idea is when you for instance would like to change the TOML support you have no idea which flag you have to set. Our flags are expressive enough and when you write them here directly you can change them directly from the console without opening some readme.

So please name them like:
```
message("          TOML_CONFIG................: " ${TOML_CONFIG})
message("          ONE_TO_MANY_ONLY...........: " ${ONE_TO_MANY_ONLY})
```